apis/v1alpha2/httproute_types.go

```diff
@@ -282,6 +283,23 @@ const (
 	HeaderMatchImplementationSpecific HeaderMatchType = "ImplementationSpecific"
 )
 
+// HTTPHeaderName is the name of an HTTP header.
```

Comment:
Do we need to say it's case-insensitive?

Reply:
We say in the place where this type is used. I'm not sure what is the right place for it. The place of use seems more correct because that gets added to the CRD. I can argue either way though.

Reply:
I think it's more appropriate to specify that it's case insensitive on the match field itself (current state). There's nothing on this type definition that can require that matching is case insensitive.

Reply:
I'm ok with anything as long as the generated documentation is clear.